### DISCUSSION TITLE
VisionFive2 u-boot

### DISCUSSION THREAD

**Discussion by treesurf** [MichaIng/DietPi#6641] [Q&A] [answered]
Using the most recent VisionFive2 u-boot
I have modified the Visionfive2 flash u-boot scripts to boot Dietpi from SD, USB and NVME. 
EMMC should also work (needs checking, I don't have a spare emmc chip).

at uboot prompt StarFive #
eraseenv
env edit bootcmd_nvme0
edit: setenv devnum 0; run nvme_boot
env edit bootcmd_usb0
edit: pci enum; usb reset; setenv devnum 0; run usb_boot
env edit bootcmd_mmc1
edit: setenv devnum 1; run mmc_boot
env edit bootcmd
edit: run distro_bootcmd
env edit boot_targets
edit: mmc1 usb0 nvme0 mmc0 dhcp
env edit kernel_comp_addr_r
edit: 0x90000000
env edit kernel_comp_size
edit: 0x10000000
saveenv

after dd image to device
edit /boot/extlinux/extlinux.conf to point to the root partition
e.g. 
usb /dev/sda1
nvme /dev/nvme0n1p1
sd /dev/mmcblk1p1
emmc /dev/mmcblk0p1

**Comment by MichaIng** [comment]
Ah yes, this looks very close to the U-Boot environment I have here:
```sh
boot_targets=nvme0 mmc0 mmc1 dhcp
bootcmd_mmc0=setenv devnum 0; run mmc_boot
bootcmd_mmc1=setenv devnum 1; run mmc_boot
bootcmd_nvme0=setenv devnum 0; run nvme_boot
```
I totally forgot about USB boot targets. I cannot see `usb_boot` in my environment. Was this added with recent U-Boot update?

The `kernel_comp_*` variables are not needed anymore with latest U-Boot, respectively I managed to get them merged into StarFive U-Boot sources.

I'm just building a kernel based on the new Linux 6.1 branch. It worked already, but did miss some features, especially the ESWIN WiFi drivers. But they have been added with today's SDK release. The package will then come with a clean default U-Boot environment and the DietPi update will offer to flash it. Also, I'll add an option to `dietpi-config` to set boot priorities.

> after dd image to device
edit /boot/extlinux/extlinux.conf to point to the root partition

This should not need to be required as the root partition is defined via PARTUUID.

**Comment by MichaIng** [comment, marked as answer]
Major update with the current DietPi beta: https://github.com/MichaIng/DietPi/issues/6664#issuecomment-1771351165
